packages/openapi-v3/src/types.ts

```diff
@@ -43,6 +43,8 @@ export interface TagsDecoratorMetadata {
   tags: string[];
 }
 
+export type VisibilityDecoratorMetadata = 'documented' | 'undocumented';
```

Comment:
@achrinza @raymondfeng 

I am not sure if `documented` is the best name, what do you think about using `public` instead?

```suggestion
export type VisibilityDecoratorMetadata = 'public' | 'undocumented';
```

Not a big deal, just an idea to consider.

Reply:
We considered `public` in the past but it can be misleading. The endpoint is still public even it does not show up in the openapi spec.